stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2523,7 +2523,7 @@ final internal class _Native${Self}StorageImpl<${TypeParameters}> {
 
   @_versioned
   internal var _capacity: Int {
-    return _body.capacity
+    return _assumeNonNegative(_body.capacity)
```

Comment:
If only there were some way to communicate this in the type system... 🤔

Which remaining assertions is this eliminating after you've moved all these checks to debug-only?

Note from my own digging in the compiler: this is lowered to value ranges on loads (fast), and not llvm.assume (slow). Good!